lib/monitoring/pub_sub.rb

```diff
@@ -0,0 +1,22 @@
+module Monitoring
+    # Module PubSub wraps ActiveSupport::Notifications, providing PubSub
+    # plumbing to custom Controllers or Collectors.
+    module PubSub
```

Comment:
It looks like you're using a module as a singleton here, which is a touch cleaner than creating a singleton class. We do need to consider a Singleton is really a global variable. We need to be careful to not create an invisible, tightly coupled dependency throughout our system.

As an alternative, this will allow you to test this singleton with unit tests, but also allow it to be used as singleton globally:
```ruby
module Monitoring
    # Module PubSub wraps ActiveSupport::Notifications, providing PubSub
    # plumbing to custom Controllers or Collectors.
    class PubSubBase
      def publish(name, payload = {}, &block)
        ActiveSupport::Notifications.instrument(name, payload, &block)
      end

      def subscribe(name)
        ActiveSupport::Notifications.subscribe(name) do |_, _, _, _, payload|
          yield payload
        end
      end

      def unsubscribe(name)
        ActiveSupport::Notifications.unsubscribe(name)
      end
    end

    class PubSub < PubSubBase
      include Singleton
    end
  end
end
```